src/lem/multiframe.rs

```diff
@@ -849,10 +849,13 @@ impl<'a, F: LurkField, C: Coprocessor<F>> nova::traits::circuit_supernova::StepC
     fn synthesize<CS: ConstraintSystem<F>>(
         &self,
         cs: &mut CS,
-        _pc: Option<&AllocatedNum<F>>,
+        pc: Option<&AllocatedNum<F>>,
         z: &[AllocatedNum<F>],
     ) -> Result<(Option<AllocatedNum<F>>, Vec<AllocatedNum<F>>), SynthesisError> {
-        let next_pc = AllocatedNum::alloc_infallible(&mut cs.namespace(|| "next_pc"), || {
```

Comment:
I think this should not even have been allocated before. The prover is allowed to non-deterministically supply *any* next `pc` — and it is each circuit's responsibility to ensure that it is capable of performing the reduction required. The necessary pc-circuit checking is now enforced by arecibo's `EnforcingStepCircuit` trait.

Reply:
Actually, scratch that — the diff confused me. This was previously an unconstrained allocation. This is the rare instance where that's actually what we intend. So it was fine before, actually.